### PR TITLE
Use the generic Set in scheduler

### LINKED ISSUE #113709: Replace all deprecated `sets.String` with `sets.Set`
In this PR https://github.com/kubernetes/kubernetes/pull/112377, `sets.Set` is newly implemented, and `sets.String` got marked as deprecated.

In the scheduler, we often use `sets.String` everywhere. This issue aims to replace all `sets.String` (+ other deprecated structs in the `sets` package) with `sets.Set` in the sig/scheduling related area.

You can refer to the following comments.
https://github.com/kubernetes/kubernetes/blob/7158e0a4ffffcf50ce7e2f6de0c25fbb414b62b0/staging/src/k8s.io/apimachinery/pkg/util/sets/string.go#L23-L24

/kind cleanup
/sig scheduling
/good-first-issue

The sig/scheduling related area (I'm not sure if all of them use `sets.String` in it.):

 - [ ] `pkg/scheduler` 
 - [ ] `test/e2e/scheduling`
 - [ ] `test/integration/scheduler`
 - [ ] `test/integration/scheduler_perf`
 - [ ] `cmd/kube-scheduler`

### PR DESCRIPTION
**Pull request by sarab97**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
This PR is for the sig/scheduling `sets.String` usage cleanup and moving over to the new generic Set type.

#### Which issue(s) this PR fixes:
Fixes #113709

#### Special notes for your reviewer:
No

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by charles-chenzz** [comment]
I think it's better to split this PR into multiple, since it changed so much file and it's hard to reivew

^just some advice

**Comment by kidddddddddddddddddddddd** [comment]
> yes, what I mean is that it's 500 lines of changed, each of them are one liner, if we can't split, at least made it multiple commit, and some tests are failed. maybe you need to check if you missed something

It's ok, but maybe consider remove/split changes in `test/e2e/framework` as this will involve other approvers.
Tests are failing because of #116946

File: test/e2e/storage/ubernetes_lite_volumes.go
Comment (kerthcet) [inline test/e2e/storage/ubernetes_lite_volumes.go:22]: Plz move this down, together with `k8s.io/*`

Comment (sarab97) [reply test/e2e/storage/ubernetes_lite_volumes.go]: done

File: test/e2e/storage/ubernetes_lite_volumes.go
Comment (kerthcet) [inline test/e2e/storage/ubernetes_lite_volumes.go:86]: Can you remove this for it will require the approval from sig storage, let's focus on sig scheduling only in the PR. Thanks and plz squash the commit. I'm LGTMed.

**Comment by kerthcet** [comment]
/retitle Use the generic Set in scheduler

File: test/e2e/storage/ubernetes_lite_volumes.go
Comment (sarab97) [reply test/e2e/storage/ubernetes_lite_volumes.go:86]: @kerthcet there is a dependency on a file form where this file and files in scheduling are using the same function. They are required to go together. If I were to revert change on this file. The same revert would be required on some of the sig/scheduling files.

Comment (sarab97) [reply test/e2e/storage/ubernetes_lite_volumes.go:86]: @kerthcet I read in docs that reviewer can add squash label and then bot can automatically squash the commits. Can you help with that.

Comment (kerthcet) [reply test/e2e/storage/ubernetes_lite_volumes.go:86]: Squash will combine the commit messages, better to squash by hand and remove the messages with no real value.

Comment (kerthcet) [reply test/e2e/storage/ubernetes_lite_volumes.go:86]: And seems everyone can add the label https://github.com/kubernetes/test-infra/blob/3364943fd7cd305a1054428efc237b9fd42201ea/label_sync/labels.md#tide/merge-method-squash

File: pkg/scheduler/apis/config/validation/validation_pluginargs.go
Comment (pohly) [inline pkg/scheduler/apis/config/validation/validation_pluginargs.go:151]: This and all other `sets.New` invocations with parameters can be simplified:

```suggestion
	supportedScheduleActions := sets.New(string(v1.DoNotSchedule), string(v1.ScheduleAnyway))
```

**Review by pohly — CHANGES_REQUESTED** [review]

File: pkg/scheduler/apis/config/validation/validation_pluginargs.go
Comment (sarab97) [reply pkg/scheduler/apis/config/validation/validation_pluginargs.go]: all done

File: pkg/scheduler/apis/config/validation/validation_pluginargs.go
Comment (pohly) [inline pkg/scheduler/apis/config/validation/validation_pluginargs.go:34]:
```suggestion
var supportedScoringStrategyTypes = sets.New(
```

`sed` didn't catch this one.

Comment (sarab97) [reply pkg/scheduler/apis/config/validation/validation_pluginargs.go:34]: oh shit. These should be limited cases only. Let me check.

Comment (sarab97) [reply pkg/scheduler/apis/config/validation/validation_pluginargs.go:34]: Looks like it was the only case

File: pkg/scheduler/apis/config/validation/validation_pluginargs.go
Comment (pohly) [inline pkg/scheduler/apis/config/validation/validation_pluginargs.go:34]: Another thought (sorry, didn't occur to me sooner): `supportedScoringStrategyTypes` doesn't need to be a set of strings, does it?

Removing all of the casting to `string` was necessary before because we only had `sets.String`. Now can create sets of arbitrary types. So perhaps try removing the casting here and in those places which use `supportedScoringStrategyTypes`?

May also apply elsewhere.

Comment (sarab97) [reply pkg/scheduler/apis/config/validation/validation_pluginargs.go:34]: didnt get you

Comment (sarab97) [reply pkg/scheduler/apis/config/validation/validation_pluginargs.go:34]: Just a thought wouldnt it be better if what you are saying is covered in different PR. As this Pr is already pretty huge.

Comment (alculquicondor) [reply pkg/scheduler/apis/config/validation/validation_pluginargs.go:34]: I think the problem is that we use the set like this `sets.List(supportedScoringStrategyTypes)` to pass it to `field.Unsupported`. Maybe we need a generic method that can cast lists into other types. But we don't have it today.

File: pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go
Comment (alculquicondor) [inline pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go:1680]: should be using New

File: pkg/scheduler/framework/plugins/nodevolumelimits/non_csi.go
Comment (alculquicondor) [inline pkg/scheduler/framework/plugins/nodevolumelimits/non_csi.go:219]: New

File: pkg/scheduler/framework/plugins/nodevolumelimits/non_csi.go
Comment (alculquicondor) [inline pkg/scheduler/framework/plugins/nodevolumelimits/non_csi.go:251]: New

File: pkg/scheduler/apis/config/validation/validation_pluginargs.go
Comment (pohly) [reply pkg/scheduler/apis/config/validation/validation_pluginargs.go:34]: That `field.Unsupported` needs a `[]string` for the valid values indeed ruins my cunning plan. Let's leave it as-is, but perhaps add a comment to `supportedScoringStrategyTypes` about "supportedScoringStrategyTypes has to be a set of strings for use with field.Unsupported`? Then the next person looking at it will know that trying to simplify it will not work and not waste time on suggesting and/or trying it.

Comment (alculquicondor) [reply pkg/scheduler/apis/config/validation/validation_pluginargs.go:34]: or somebody could introduce a `list` package that knows how to do type conversions :)

```golang
package list

func Cast[A, B](in []A) ([]B) {
```

File: pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go
Comment (sarab97) [reply pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go:1680]: like this?
```suggestion
			deletedPodNames := sets.New[string]()
			patchedPodNames := sets.New[string]()
```

Comment (alculquicondor) [reply pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go:1680]: yes

**Review by alculquicondor — COMMENTED** [review]
/lgtm

**Comment by kerthcet** [comment]
Can you squash the commit again @sarab97 , the commit is duplicated.

**Comment by sarab97** [comment]
@kerthcet done

**Pull request merged**
